test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -38,6 +38,60 @@ public MetricApiTest(ITestOutputHelper output)
             this.output = output;
         }
 
+        [Theory]
+        [InlineData(AggregationTemporality.Cumulative)]
+        [InlineData(AggregationTemporality.Delta)]
+        public void StreamNamesDuplicatesAreNotAllowedTest(AggregationTemporality temporality)
+        {
+            var metricItems = new List<Metric>();
+            int metricCount = 0;
+            var metricExporter = new TestExporter<Metric>(ProcessExport);
```

Comment:
Could we use InMemoryExporter?

Main reason is that we build InMemoryExporter for folks to use in their unit tests, I think we need to use it in our own tests, too.

Reply:
Yes. Once https://github.com/open-telemetry/opentelemetry-dotnet/issues/2361 is fixed.

Reply:
Does #2361 block this change?

Reply:
no. this particular test is not blocked. 
Want to switch all the test in this class to use InMemoryExporter together once #2361 is ready.